### DISCUSSION TITLE
Reacting to changes in test data files/Making watch mode posible on non code files

### DISCUSSION THREAD

**Discussion by jkavalec** [vitest-dev/vitest#5864] [General]
Recently, I was doing lots of tests on text files, there is an inconvenience where you have to touch spec file to rerun the test in watch mode if you change test data file.

First thing that I tried was to add txt files into include field within Vitest config, and then tried to log batch of files within Reporter, this didn't work and these txt files are for Vitest invisible and only Typescript files relevant to tests are logged.

Is there any way that would allow me to get notified when any file with any pattern changes?

**Comment by hi-ogawa** [comment]
Vitest uses Vite's module graph to watch change and trigger test rerun, so if you're reading text files with `fs.readFile("some-file.txt")` then such "dependency" is not recorded in module graph.

If you want to rerun all tests, then you can add text files to `forceRerunTriggers` https://vitest.dev/config/#forcereruntriggers

Another approach (I haven't checked though) would be to use Vite's import mechanism to load text file so that they also join in module graph. For example, if you use `import someData from "./some-file.txt?raw"` or `import.meta.glob("./some-dir/*.txt", { query: "?raw" )` inside test files, then that'll trigger test rerun when changing text files.
(More advanced technique would be to write a plugin to use `this.addWatchFile`, which should probably have a similar effect on test rerun.)

**Reply by jkavalec** [reply]
Thank you for giving some good advice, will be looking into how to handle this on Vite side with addWatchFile.

**Reply by fregante** [reply]
`import.meta.glob` seems to be the answer. It accepts arrays and paths to non-JS files, so it's perfect for this.
